samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -220,6 +225,13 @@ def __enter__(self) -> "InvokeContext":
 
         self._stacks = self._get_stacks()
 
+        try:
+            _, matches_output = get_lint_matches(self._template_file, self._verbose, self._aws_region)
+            if matches_output:
+                LOG.warning("Lint Error found, containter creation might fail: %s", matches_output)
+        except UserException as e:
```

Comment:
Can we make sure everything is wrapped as `UserException` so there won't be a new exception type which might be thrown from `get_lint_matches`?

Reply:
errors within cfn-lint seem to be caught and wrapped as `CfnLintExitException` which is handled in this method. I think it should be fine.